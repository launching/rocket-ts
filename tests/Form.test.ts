import Vue from 'vue';
import ElementUI from 'element-ui';

import { shallowMount, mount } from '@vue/test-utils';
import RForm from '../src/components/Form/Form.vue';

Vue.use(ElementUI);

describe('RForm.vue', () => {
  const children = [
    {
      label: '密码',
      widget: 'password',
      name: 'pass',
    },
    {
      label: '确认密码',
      widget: 'password',
      name: 'checkPass',
    },
    {
      label: '年龄',
      widget: 'number',
      name: 'age',
    },
    {
      label: '性别',
      widget: 'radio',
      name: 'sex'
    }
  ];

  it('render form', async () => {
    const wrapper = mount(RForm, {
      propsData: {
        children,
        defaultModel: () => { 
          return Promise.resolve({pass: '123',
          checkPass: '1234',
          age: '12',
          sex: '1'})
        }
      },
    });
    await wrapper.vm.$nextTick();

    expect(wrapper.html()).toMatchSnapshot();

    wrapper.find('input[type="password"]').setValue('123')
    await wrapper.vm.$nextTick();

    const vm:any = wrapper.vm as RForm ;
    expect(vm.model.pass).toEqual('123');
  });
});
