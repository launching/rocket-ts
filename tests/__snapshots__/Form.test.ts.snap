// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RForm.vue render form 1`] = `
<div class="r-form">
  <form class="el-form" children="[object Object],[object Object],[object Object],[object Object]" defaultmodel="function () {
                                return Promise.resolve({ pass: '123',
                                    checkPass: '1234',
                                    age: '12',
                                    sex: '1' });
                            }">
    <div class="el-form-item"><label for="pass" class="el-form-item__label">密码</label>
      <div class="el-form-item__content">
        <div class="r-input" label="密码" widget="password">
          <div class="el-input">
            <!----><input type="password" autocomplete="off" class="el-input__inner">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <div class="r-toolbar"></div>
        </div>
        <!---->
      </div>
    </div>
    <div class="el-form-item"><label for="checkPass" class="el-form-item__label">确认密码</label>
      <div class="el-form-item__content">
        <div class="r-input" label="确认密码" widget="password">
          <div class="el-input">
            <!----><input type="password" autocomplete="off" class="el-input__inner">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <div class="r-toolbar"></div>
        </div>
        <!---->
      </div>
    </div>
    <div class="el-form-item"><label for="age" class="el-form-item__label">年龄</label>
      <div class="el-form-item__content">
        <div class="r-input" label="年龄" widget="number">
          <div class="el-input">
            <!----><input type="password" autocomplete="off" class="el-input__inner">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <div class="r-toolbar"></div>
        </div>
        <!---->
      </div>
    </div>
    <div class="el-form-item"><label for="sex" class="el-form-item__label">性别</label>
      <div class="el-form-item__content">
        <div role="radiogroup" class="el-radio-group" label="性别" widget="radio"></div>
        <!---->
      </div>
    </div>
    <div class="el-form-item">
      <!---->
      <div class="el-form-item__content">
        <div class="r-toolbar"></div>
        <!---->
      </div>
    </div>
  </form>
</div>
`;
